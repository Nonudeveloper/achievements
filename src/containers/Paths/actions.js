export const PATHS_OPEN = "PATHS_OPEN";
export const pathsOpen = () => ({
  type: PATHS_OPEN
});

export const PATH_GAPI_AUTHORIZED = "PATH_GAPI_AUTHORIZED";
export const pathGAPIAuthorized = status => ({
  type: PATH_GAPI_AUTHORIZED,
  status
});

export const PATH_DIALOG_SHOW = "PATH_DIALOG_SHOW";
export const pathDialogShow = pathInfo => ({
  type: PATH_DIALOG_SHOW,
  pathInfo
});

export const PATH_PROBLEM_DIALOG_SHOW = "PATH_PROBLEM_DIALOG_SHOW";
export const pathProblemDialogShow = problemInfo => ({
  type: PATH_PROBLEM_DIALOG_SHOW,
  problemInfo
});

export const PATHS_JOINED_FETCH_SUCCESS = "PATHS_JOINED_FETCH_SUCCESS";
export const pathsJoinedFetchSuccess = paths => ({
  type: PATHS_JOINED_FETCH_SUCCESS,
  paths
});

export const PATH_DIALOG_HIDE = "PATH_DIALOG_HIDE";
export const pathDialogHide = () => ({
  type: PATH_DIALOG_HIDE
});

export const PATH_CHANGE_REQUEST = "PATH_CHANGE_REQUEST";
export const pathChangeRequest = pathInfo => ({
  type: PATH_CHANGE_REQUEST,
  pathInfo
});

export const PATH_CHANGE_SUCCESS = "PATH_CHANGE_SUCCESS";
export const pathChangeSuccess = (pathInfo, pathKey) => ({
  type: PATH_CHANGE_SUCCESS,
  pathInfo,
  pathKey
});

export const PATH_SELECT = "PATH_SELECT";
export const pathSelect = pathId => ({
  type: PATH_SELECT,
  pathId
});

export const PATH_PROBLEM_MOVE_REQUEST = "PATH_PROBLEM_MOVE_REQUEST";
export const pathProblemMoveRequest = (pathId, problemId, direction) => ({
  type: PATH_PROBLEM_MOVE_REQUEST,
  pathId,
  problemId,
  direction
});

export const PATH_PROBLEM_MOVE_SUCCESS = "PATH_PROBLEM_MOVE_SUCCESS";
export const pathProblemMoveSuccess = (pathId, problemId, direction) => ({
  type: PATH_PROBLEM_MOVE_SUCCESS,
  pathId,
  problemId,
  direction
});

export const PATH_PROBLEM_MOVE_FAIL = "PATH_PROBLEM_MOVE_FAIL";
export const pathProblemMoveFail = (pathId, problemId, direction, reason) => ({
  type: PATH_PROBLEM_MOVE_FAIL,
  pathId,
  problemId,
  direction,
  reason
});

export const PATH_PROBLEM_CHANGE_REQUEST = "PATH_PROBLEM_CHANGE_REQUEST";
export const pathProblemChangeRequest = (pathId, problemInfo) => ({
  type: PATH_PROBLEM_CHANGE_REQUEST,
  pathId,
  problemInfo
});

export const PATH_PROBLEM_CHANGE_SUCCESS = "PATH_PROBLEM_CHANGE_SUCCESS";
export const pathProblemChangeSuccess = (pathId, problemInfo, problemKey) => ({
  type: PATH_PROBLEM_CHANGE_SUCCESS,
  pathId,
  problemInfo,
  problemKey
});

export const PATH_PROBLEM_CHANGE_FAIL = "PATH_PROBLEM_CHANGE_FAIL";
export const pathProblemChangeFail = (pathId, problemInfo, reason) => ({
  type: PATH_PROBLEM_CHANGE_FAIL,
  pathId,
  problemInfo,
  reason
});
