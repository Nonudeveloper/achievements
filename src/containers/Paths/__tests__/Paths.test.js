import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import sinon from "sinon";

import { Paths } from "../Paths";

describe("<Paths.test.js>", () => {
  let shallow;
  let dispatch;

  beforeEach(() => {
    shallow = createShallow();
    dispatch = sinon.spy();
  });

  it("should check snapshot", () => {
    const wrapper = shallow(
      <Paths
        dispatch={dispatch}
        joinedPaths={{}}
        myPaths={{}}
        problems={[]}
        publicPaths={{}}
        selectedPathId="asd"
        ui={{ dialog: {} }}
        uid={"abcd"}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
