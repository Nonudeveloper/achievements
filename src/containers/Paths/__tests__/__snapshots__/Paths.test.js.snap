// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paths.test.js> should check snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Paths
    dispatch={[Function]}
    joinedPaths={Object {}}
    myPaths={Object {}}
    problems={Array []}
    publicPaths={Object {}}
    selectedPathId="asd"
    ui={
        Object {
            "dialog": Object {},
          }
    }
    uid="abcd"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <WithStyles(Breadcrumbs)
          paths={
                    Array [
                              Object {
                                "label": "Paths",
                              },
                            ]
          }
/>,
        <PathTabs
          dispatch={[Function]}
          joinedPaths={Object {}}
          myPaths={
                    Object {
                              "abcd": Object {
                                "name": "Default",
                              },
                            }
          }
          publicPaths={Object {}}
/>,
        <PathDialog
          dispatch={[Function]}
          open={false}
          path={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "paths": Array [
            Object {
              "label": "Paths",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dispatch": [Function],
          "joinedPaths": Object {},
          "myPaths": Object {
            "abcd": Object {
              "name": "Default",
            },
          },
          "publicPaths": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dispatch": [Function],
          "open": false,
          "path": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(Breadcrumbs)
            paths={
                        Array [
                                    Object {
                                      "label": "Paths",
                                    },
                                  ]
            }
/>,
          <PathTabs
            dispatch={[Function]}
            joinedPaths={Object {}}
            myPaths={
                        Object {
                                    "abcd": Object {
                                      "name": "Default",
                                    },
                                  }
            }
            publicPaths={Object {}}
/>,
          <PathDialog
            dispatch={[Function]}
            open={false}
            path={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "paths": Array [
              Object {
                "label": "Paths",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dispatch": [Function],
            "joinedPaths": Object {},
            "myPaths": Object {
              "abcd": Object {
                "name": "Default",
              },
            },
            "publicPaths": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dispatch": [Function],
            "open": false,
            "path": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
